File: backend/src/main/java/com/minetec/backend/dto/info/VehicleInfo.java
package com.minetec.backend.dto.info;

import lombok.Data;

import javax.validation.constraints.NotNull;
import java.util.List;
import java.util.UUID;

@Data
public class VehicleInfo implements Comparable<VehicleInfo> {

    @NotNull
    private UUID uuid;
    private SiteInfo siteInfo;
    private VehicleTypeInfo vehicleType;
    private String fleetNo;
    private String vinNo;
    private String serialNo;
    private String unit;
    private boolean isUsable;
    private String registrationNo;
    private String currentMachineHours;
    private String serviceIntervalHours;
    private String lastServiceDate;
    private String lastServiceHours;
    private boolean hasTask;
    private List<ImageInfo> imageInfos;
    private Integer remainingHours;
    private String serviceWarning;

    @Override
    public int compareTo(final VehicleInfo o) {
        return this.getRemainingHours().compareTo(o.getRemainingHours());
    }
}

File: frontend/src/app/models/vehicle/vehicle-info.ts
import { Image } from '../general/image';
import { SiteForm } from '../sites/site-form';

export class VehicleInfo {
    public vehicleDetailTypeAttrInfos: [];
    public vehicleInfo: Vehicle;
    public vehicleTypeInfo: VehicleTypeInfo;
    public imageInfos: Image[];
}
export class VehicleTypeInfo {
    public uuid: string;
    public vehicleTypeAttributes: string;
    public vehicleTypeDesc: string;

}
export class Vehicle {

    public uuid: string;
    public fleetNo: string;
    public serialNo: string;
    public registrationNo: string;
    public vinNo: string;
    public currentMachineHours: string;
    public serviceIntervalHours: string;
    public lastServiceDay: string;
    public lastServiceHours: string;
    public vehicleTypeUuid: string;
    public vehicleType: VehicleType;
    public imageUuid: string;
    public siteInfo: SiteForm;
    public imageInfos: Image[];
    public remainingHours: string;
    public serviceWarning: string;

}

export class VehicleType {
    public uuid: string;
    public name: string;
    public vehicleTypeAttributes: VehicleTypeAttributes[];
}
export class VehicleTypeAttributes {
    public uuid: string;
    public selectedVehicleAttrUuid: string;
    public vehicleAttributeName: string;
    public vehicleAttributeValues: VehicleAttributeValues[];
}


export class VehicleAttributeValues {
    public uuid: string;
    public vehicleAttributeValue: string;
}

export class VehicleImage {
    public vehicleUuid: string;
    public imageInfos: Image[];
}
